crates/matrix-sdk-crypto/src/error.rs

```diff
@@ -141,6 +141,19 @@ pub enum MegolmError {
     /// The storage layer returned an error.
     #[error(transparent)]
     Store(#[from] CryptoStoreError),
+
```

Comment:
It really feels like we should create a separate and more specific error type instead of adding ever-more variants here. Not needed for this PR but please put something on the board.

Reply:
While I was investigating https://github.com/matrix-org/matrix-rust-sdk/pull/3701#discussion_r1701500823 I noticed that `matrix_sdk_common::deserialized_responses::VerificationLevel` maps pretty well to what we want here, except that it's missing a level, so I added the level and referenced it here.